lib/finch/telemetry.ex

```diff
@@ -63,6 +63,7 @@ defmodule Finch.Telemetry do
 
   #### Metadata
 
+    * `:name` - The name of the Finch instance.
```

Comment:
Thanks for this contribution! We should probably mention that name will only be populated for HTTP/1, but the more I think about it, I'd really like to get H2 done all at the same time.

Reply:
Hey, appreciate your review! I've included a note indicating that name is populated for HTTP/1 only in events utilised for both HTTP/1 and HTTP/2.

I believe beginning with the HTTP/1 change is a great step. It's a simple and direct modification that enhances the descriptiveness and usefulness of metrics. Plus, it can be easily expanded to HTTP/2 when the need arises.